### DISCUSSION TITLE
Is Next.js suitable for project with complex back-end ?

### DISCUSSION THREAD

**Discussion by clemclx** [vercel/next.js#18365] [Help]
Hey all,

I am facing troubles to find resources about next.js and back-end management.
I know that it is possible to create API routes with the pages folder but my question is : Is Next.js really suitable for full-stack project with some business logic in the back ?

I mean back-end connected to DB, with user authentification, websocket..

Can someone help me with that question ?

Thanks for reading

**Comment by GoEagles999** [comment]
Next.js is just that.. a client-side React framework so it shouldn’t limit you from accomplishing anything you want on the server!

**Comment by ljosberinn** [comment]
The question is - why would it not be? Take a look [at the showcases here](https://nextjs.org/showcase) and you'll see a variety of complexity involved.

**Reply by fahadali32** [reply]
thanks sir i am also a beginer developer or student.i wanted to make complex app but you clear my doubt thanks

**Comment by revskill10** [comment]
Yes, it's possible, but don't deploy it as a serverless app. Use it to replace express app you're using.

**Reply by ShlokDesai33** [reply]
Could you elaborate on your answer further?

**Reply by GabenGar** [reply]
He means you have to self-host if you want to migrate an express app. This has its own can of worms and NextJS is not going to replace the proper express server entirely.
